test/integ/test_cursor.py

```diff
@@ -1195,3 +1203,74 @@ def test__log_telemetry_job_data(conn_cnx, caplog):
         logging.WARNING,
         "Cursor failed to log to telemetry. Connection object may be None.",
     ) in caplog.record_tuples
+
+
+@pytest.mark.skipolddriver(reason="new feature in v2.5.0")
+@pytest.mark.parametrize(
+    "result_format,expected_chunk_type",
+    (
+        ("json", JSONResultBatch),
+        ("arrow", ArrowResultBatch),
+    ),
+)
+def test_resultbatch(
+    conn_cnx,
+    result_format,
+    expected_chunk_type,
+):
+    """This test checks the following things:
+    1. After executing a query can we pickle the result batches
```

Comment:
Maybe there is one more thing to test or demo: pre_pickle does not contain the actual data, the lazy fetching happens as expected(when iterating)

Reply:
> pre_pickle does not contain the actual data

`ResultBatch` es never contain data (other than the embedded first chunk), not even after you fetch them once.

Let me see how hard testing the lazy fetching would be 👍 I can probably do some `Mock` magic to verify it automatically.

Reply:
> pre_pickle does not contain the actual data

A `ResultBatch` doesn't cache its results, because they way we process them can differ. For example see the code for `fetch_arrow_batches` and `fetchone` when the result is in Arrow format.

Added a test for checking lazy fetching correctness!